JM/CF/Scripts/1_Core/CommunityFramework/Files/CF_File.c

```diff
@@ -0,0 +1,167 @@
+class CF_File : Managed
+{
+	protected string m_FileName;
+	protected string m_Folder;
+	protected bool m_IsDirectory;
+	protected bool m_IsHidden;
+	protected bool m_IsReadOnly;
+	protected bool m_IsValid;
+
+	/**
+	 * @param pattern The search pattern
+	 * @param [inout] files An array where the list of files will be written to
+	 * @param flags If it should be looked in directories or .pak files, or both
+	 * 
+	 * @return If the pattern was valid
+	 */
+	static bool Find(string pattern, inout array<ref CF_File> files, FindFileFlags flags = 2)
+	{
+#ifdef CF_TRACE_ENABLED
+		auto trace = CF_Trace_3("CF_File", "Find").Add(pattern).Add(files).Add(flags);
+#endif
+
+		if (!files)
+		{
+			files = new array<ref CF_File>();
+		}
+
+		string fileName;
+		FileAttr fileAttr;
+		FindFileHandle handle = FindFile(pattern, fileName, fileAttr, flags);
+		if (!handle)
+		{
+			return false;
+		}
```

Comment:
Please double check that handle is only NULL when the pattern was invalid. Try to search for a file that does not exist and see if handle is valid. I suspect it might be NULL as well, in which case we can not tell if the pattern was correct or not. In that case have the function return void.

Reply:
Returns false with invalid pattern.

Using the below code with pattern "$profile:CommunityOnlineTools/*" and "$profile:SomeFolderDoesntExist/*", the first returns true, the second returns false.

```
array<ref CF_File> files;
string pattern = "$profile:CommunityOnlineToolcs/*";

bool success = CF_File.Find(pattern, files);
Print(success);
foreach (auto file : files)
{
	Print(file.GetDebugName());
}
```